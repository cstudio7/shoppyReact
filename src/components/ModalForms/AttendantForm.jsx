import React, { Component, Fragment } from 'react';
import { Button, Form } from 'semantic-ui-react';
import './ModalForms.scss';

export default class AttendantForm extends Component {
  displayFormContent = () => {
    const { nameToDisplay, imageUrl, description, mobileNumber } = this.props.details;
    return (
      <Fragment>
        <Form.Field className="formField">
          <input placeholder="Email" typeof="email" defaultValue={description} />
        </Form.Field>
        <Form.Field className="formField">
          <input placeholder="Password" typeof="password" />
        </Form.Field>
        <Form.Field className="formField">
          <input placeholder="Attendant Name" defaultValue={nameToDisplay} />
        </Form.Field>
        <Form.Field className="formField">
          <input placeholder="Mobile Number" typeof="tel" defaultValue={mobileNumber} />
        </Form.Field>
        <Form.Field className="formField">
          <input
            placeholder="Image URL (example:https://exampleimageurl.jpg)"
            typeof="url"
            defaultValue={imageUrl}
          />
          <a target="_blank" rel="noopener noreferrer" href="https://imgur.com/">
            Upload image here
          </a>
        </Form.Field>
      </Fragment>
    );
  };

  render() {
    return (
      <Form className="modalForm">
        {this.displayFormContent()}
        <Button type="submit">Submit</Button>
      </Form>
    );
  }
}
