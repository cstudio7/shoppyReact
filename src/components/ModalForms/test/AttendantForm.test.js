import React from 'react';
import { shallow } from 'enzyme';
import AttendantForm from '../AttendantForm';

describe('Test the AttendantForm Component', () => {
  it('should match snapshot', () => {
    const details = {
      nameToDisplay: 'John Doe',
      imageUrl: 'https://i.imgur.com/27DUH5b.jpg',
      description: 'admin@mail.com',
      mobileNumber: '08076543241'
    };

    const wrapper = shallow(<AttendantForm details={details} />);
    expect(wrapper).toMatchSnapshot();
  });
});
