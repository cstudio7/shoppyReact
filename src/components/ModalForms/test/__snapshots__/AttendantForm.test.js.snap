// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the AttendantForm Component should match snapshot 1`] = `
<Form
  as="form"
  className="modalForm"
  onSubmit={[Function]}
>
  <FormField
    className="formField"
  >
    <input
      name="email"
      onChange={[Function]}
      placeholder="Email"
      typeof="email"
      value=""
    />
  </FormField>
  <FormField
    className="formField"
  >
    <input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      typeof="password"
      value=""
    />
  </FormField>
  <FormField
    className="formField"
  >
    <input
      name="first_name"
      onChange={[Function]}
      placeholder="Attendant Name"
      value=""
    />
  </FormField>
  <FormField
    className="formField"
  >
    <input
      name="mobile_number"
      onChange={[Function]}
      placeholder="Mobile Number"
      typeof="tel"
      value=""
    />
  </FormField>
  <FormField
    className="formField"
  >
    <input
      name="image_url"
      onChange={[Function]}
      placeholder="Image URL (example:https://exampleimageurl.jpg)"
      typeof="url"
      value=""
    />
    <a
      href="https://imgur.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Upload image here and provide link
    </a>
  </FormField>
  <Button
    as="button"
    type="submit"
  >
    Submit
  </Button>
</Form>
`;
